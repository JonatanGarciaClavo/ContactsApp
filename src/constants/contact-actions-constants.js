export const REQUEST_CONTACT_SUCCESS = 'REQUEST_CONTACT_SUCCESS';
export const UPDATE_CONTACT_ATTRIBUTE = 'UPDATE_CONTACT_ATTRIBUTE';
export const VALIDATE_CONTACT_ATTRIBUTE = 'VALIDATE_CONTACT_ATTRIBUTE';
export const VALIDATE_CONTACT = 'VALIDATE_CONTACT';
export const REQUEST_CONTACT = 'REQUEST_CONTACT';
export const REQUEST_SAVE_CONTACT = 'REQUEST_SAVE_CONTACT';
export const TRANSTION_TO_EDIT_CONTACT = 'TRANSTION_TO_EDIT_CONTACT';
export const RESET_CONTACT = 'RESET_CONTACT';
