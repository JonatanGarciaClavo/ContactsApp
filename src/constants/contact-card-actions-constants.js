export const RESET_CONTACT_CARD = 'RESET_CONTACT_CARD';
export const REQUEST_CONTACT_CARD_SUCCESS = 'REQUEST_CONTACT_CARD_SUCCESS';
export const REQUEST_CONTACT_CARD = 'REQUEST_CONTACT_CARD';
export const REQUEST_DELETE_CONTACT_CARD = 'REQUEST_DELETE_CONTACT_CARD';
export const TRANSTION_TO_EDIT_CONTACT_CARD = 'TRANSTION_TO_EDIT_CONTACT_CARD';
