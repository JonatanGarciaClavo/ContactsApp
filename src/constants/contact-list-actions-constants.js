export const LOADING_CONTACT_LIST = 'LOADING_CONTACT_LIST';
export const INITIALIZE_CONTACT_LIST = 'INITIALIZE_CONTACT_LIST';
export const CHANGE_LIST_MODE = 'CHANGE_LIST_MODE';
