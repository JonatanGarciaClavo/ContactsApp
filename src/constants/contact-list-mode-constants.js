export const LIST_MODE = 'list';
export const CARD_MODE = 'card'
