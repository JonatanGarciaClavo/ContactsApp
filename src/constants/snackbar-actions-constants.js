export const CLOSE_ERROR_MESSAGE = 'CLOSE_ERROR_MESSAGE';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
