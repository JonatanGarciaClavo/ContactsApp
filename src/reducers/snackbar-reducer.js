import { CLOSE_ERROR_MESSAGE, SET_ERROR_MESSAGE } from '../constants/snackbar-actions-constants';

export const initialState = {
  open: false,
  message: 'Unknown error',
};

const snackbar = (state = initialState, action) => {
  switch (action.type) {
    case (CLOSE_ERROR_MESSAGE):
      return { ...initialState };
    case (SET_ERROR_MESSAGE):
      return { ...state, open: true, message: action.message };
    default:
      return state;
  }
};

export default snackbar;
