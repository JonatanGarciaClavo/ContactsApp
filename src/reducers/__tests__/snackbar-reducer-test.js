import { initialState, default as snackbar } from '../snackbar-reducer';
import { CLOSE_ERROR_MESSAGE, SET_ERROR_MESSAGE } from '../../constants/snackbar-actions-constants';

describe('Test snackbar reducer', () => {
  it('returns the initialState state on an undefined state', () => {
    expect(snackbar(undefined, { type: '_NULL' })).toMatchSnapshot();
  });
  it('returns the same state on an unhandled action', () => {
    expect(snackbar(initialState, { type: '_NULL' })).toMatchSnapshot();
  });

  it('handles close snackbar action', () => {
    expect(snackbar(initialState, { type: CLOSE_ERROR_MESSAGE })).toMatchSnapshot();
  });

  it('handles display error action', () => {
    const customError = 'Custom error';
    expect(snackbar(initialState, { type: SET_ERROR_MESSAGE, message: customError }))
      .toMatchSnapshot();
  });
});
