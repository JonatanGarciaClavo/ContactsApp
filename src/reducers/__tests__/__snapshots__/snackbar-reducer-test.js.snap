exports[`Test snackbar reducer handles close snackbar action 1`] = `
Object {
  "message": "Unknown error",
  "open": false,
}
`;

exports[`Test snackbar reducer handles display error action 1`] = `
Object {
  "message": "Custom error",
  "open": true,
}
`;

exports[`Test snackbar reducer returns the initialState state on an undefined state 1`] = `
Object {
  "message": "Unknown error",
  "open": false,
}
`;

exports[`Test snackbar reducer returns the same state on an unhandled action 1`] = `
Object {
  "message": "Unknown error",
  "open": false,
}
`;
